### ISSUE TITLE
A bug when using union operator

### ISSUE THREAD

**Issue by D-Benjamin** [openmc-dev/openmc#1170]
Hi All,
    I may find a bug when using union operator("|"). 

```
 fuel_region = -r1 & +z0 & -z3
 fuel_region |= (+z0 & -z1 & +r1 & -r2)
 cell = openmc.Cell(1, fill = fuel, region = fuel_region)
```

   When I using "|" to define a region, openmc.run() will occur following warning:

`WARNING: Particle xxx underwent maximum number of event.`

 But, if I change another define method( Such as separate this region to two region, rename it ),
 this warning will not happen again. I guess it may a buy, hope to get you reply!

```
 region1 = -r1 & +z0 & -z3
 region2 = +z0 & -z1 & +r1 & -r2
 cell1 = openmc.Cell(1, fill = fuel, region = region1)
 cell2 = openmc.Cell(2. fill = fuel, region = region2)
```

    Best wishes!
    Benjamin

**Comment by paulromano** [comment]
@D-Benjamin Thank for reporting this. Is it possible to send a full example script that results in the behavior you're seeing? Also, what version of OpenMC are you running?

**Comment by makeclean** [comment]
This may just be me @D-Benjamin, but I think your syntax isn't correct

You have 
```
fuel_region = -r1 & +z0 & -z3
fuel_region |= (+z0 & -z1 & +r1 & -r2)
cell = openmc.Cell(1, fill = fuel, region = fuel_region)
```
I think what you want is 
```
fuel_region = -r1 & +z0 & -z3
fuel_region += |(+z0 & -z1 & +r1 & -r2)
cell = openmc.Cell(1, fill = fuel, region = fuel_region)
```
I think in your first case python is interpreting the | as a bitwise or operation.

**Comment by D-Benjamin** [comment]
> @D-Benjamin Thank for reporting this. Is it possible to send a full example script that results in the behavior you're seeing? Also, what version of OpenMC are you running?

Hi Paul,
    Thanks for your reply and here is the script. My OpenMC version is v.0.10.0

**Comment by D-Benjamin** [comment]
> This may just be me @D-Benjamin, but I think your syntax isn't correct
> 
> You have
> 
> ```
> fuel_region = -r1 & +z0 & -z3
> fuel_region |= (+z0 & -z1 & +r1 & -r2)
> cell = openmc.Cell(1, fill = fuel, region = fuel_region)
> ```
> I think what you want is
> 
> ```
> fuel_region = -r1 & +z0 & -z3
> fuel_region += |(+z0 & -z1 & +r1 & -r2)
> cell = openmc.Cell(1, fill = fuel, region = fuel_region)
> ```
> I think in your first case python is interpreting the | as a bitwise or operation.

Hi Davis,
Thanks for your reply, but I think there isn't a syntax error about "fuel_region |= (+z0 & -z1 & +r1 & -r2)", cause there is a similar syntax in the [example notebook](https://openmc.readthedocs.io/en/v0.10.0/examples/candu.html).

`water_cells[i].region &= +pin_boundary`

Also, I test it, python doesn't have the syntax

`fuel_region += |(+z0 & -z1 & +r1 & -r2)`

And, "+=" will be considered as "&=" in the OpenMC syntax.
Anyway, I really appreciate for your cordial reply and welcome to your reply if your have some another idea.

**Comment by t2015q** [comment]
Hello,I also met this problem  :

WARNING: Particle xxx underwent maximum number of event

But  I think you can try  this:

point = openmc.stats.Point((0, 0, 0))

**Comment by D-Benjamin** [comment]
@t2015q 
Welcome to your comment. But I'm not understand what do you mean. (0, 0, 0) is a point on boundary surface of my model, the way as you say will cause the error:
```
ERROR: More than 95% of external source sites sampled were rejected. Please
        check your external source definition.
```

**Comment by t2015q** [comment]
Before I encountered the same problem as you, I just adjusted the coordinates of the Point until I found one that could run openmc. And I don't why, sorry.

**Comment by paulromano** [comment]
This should be fixed now with the merging of #1172.
